app/models/statistics_announcement.rb

```diff
@@ -25,7 +25,10 @@ def can_publish_to_publishing_api?
   belongs_to :publication
 
   has_one  :current_release_date,
-           -> { order("created_at DESC") },
+           lambda {
+             joins(:statistics_announcement)
```

Comment:
I think this should be:

```suggestion
             joins(:statistics_announcement_dates)
```

right?

Reply:
So this is the pattern you established in documents here 

https://github.com/alphagov/whitehall/blob/main/app/models/document.rb#L27-L30

When you update it to your suggestion you get

<img width="1091" alt="image" src="https://github.com/alphagov/whitehall/assets/42515961/7dd1aa30-837d-45a9-ad59-1b6b5611b2fa">

Reply:
If fact removing inverse_of while less performant would remove the need to reload before a touching the statistics announcement again 🤔

Reply:
Oh right, I see, my bad.

I was confused because it looked like we were trying to do `StatisticsAnnouncement.joins(:statistics_announcement)` and I was wondering why it was working. But I'd missed the `class_name` bit.

Reply:
Re: using `inverse_of` and needing to `reload`. I wasn't 100% clear why reload was needed to touch? 🤔 

For what it's worth, the Document model joins `latest_edition` using the `inverse_of` option too.

https://github.com/alphagov/whitehall/blob/2694ef04a54098144998dd28c75e48e4f1a12c15/app/models/document.rb#L27-L30

Reply:
Yea i shamelessly copied your implementation 😂 basically the inverse_of means that it uses the preloaded object that has since been updated and is now stale. So when it updates the publication from draft to published it isn't able to see that change in the assoicated document (stale one)